server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueView.java

```diff
@@ -86,7 +86,9 @@ public MailQueueView create(MailQueueName mailQueueName) {
 
     @Override
     public long getSize() {
-        return Iterators.size(browse());
+        return Iterators.size(cassandraMailQueueBrowser.browseReferences(mailQueueName)
+                .join()
+                .iterator());
```

Comment:
as you have a Stream, you should be able to write:
```
    return cassandraMailQueueBrowser.browseReferences(mailQueueName)
            .join()
            .count();
```
no?

Reply:
so right